### ISSUE TITLE
Tests failing after installation

### ISSUE THREAD

**Issue by RalphTheBest** [hpcugent/easybuild-framework#1719]
Hello,

I've recently installed EasyBuild in my machine using the bootstrap script and it was apparently successful. However, when I run the unit test indicated in the installation guide, I got one of the tests failed and I'm not able to find the reason. So if anyone has any idea why this may be, I would really appreciate the hint.

I'm putting here links to the log and the stdout of the command `python -m test.framework.suite` : 
**stdout**
[https://drive.google.com/open?id=0B0eJDPNaAh6BalE0SS1McTEtUVE](url)
**Log file**
[https://drive.google.com/open?id=0B0eJDPNaAh6BTTh6MjZTeHR4enM](url)

The result of the environment commands:

```
$python -V
Python 2.7.6

$type module
module is a function                                                                               
module ()                                                                                          
{                                                                                                  
    eval $($LMOD_CMD bash "$@");                                                                   
    [ $? = 0 ] && eval $(${LMOD_SETTARG_CMD:-:} -s sh)                                             
}                       
                                                                           
$ type -f module                                       
bash: type: module: not found                
                                                      
$module --version                                     
                                                                                                   
Modules based on Lua: Version 6.0.1  2015-06-10 16:45                                              
    by Robert McLay mclay@tacc.utexas.edu                                                          
                                                                                                   
$ module av EasyBuild
                                                                                                   
----------------------------- /home/adrian/Apps/easybuild/modules/all -----------------------------
   EasyBuild/2.7.0                                                                                 
                                                                                                   
Use "module spider" to find all possible modules.                                                  
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".

 $which -a eb                                          
/home/adrian/Apps/easybuild/software/EasyBuild/2.7.0/bin/eb                                        

$ eb --version
This is EasyBuild 2.7.0 (framework: 2.7.0, easyblocks: 2.7.0) on host adrian-OptiPlex-780.
```

Thanks for your time and your help.

**Comment by boegel** [comment]
@RalphTheBest I think your problem is caused by a bug in the Lmod version you're using. 

Please upgrade your Lmod installation to a version more recent than 6.0.1, available from either https://sourceforge.net/projects/lmod/files/ or https://github.com/TACC/Lmod/releases (I would recommend the latter).

I will update http://easybuild.readthedocs.org/en/latest/Installing-Lmod-without-root-permissions.html#installing-lmod to mention a newer version.

**Comment by boegel** [comment]
@RalphTheBest let us know in case upgrading Lmod doesn't fix your problem

**Comment by RalphTheBest** [comment]
Thanks for the hint, but unfortunately is still failing. In fact, last time it said one error after the test, right now I'm getting three, so I have no idea what can be wrong.
Again, I leave the links to the log and stdout of the test:
**Stdout**
https://drive.google.com/open?id=0B0eJDPNaAh6Bb256V3ZBM3JLeHM
**Log**
https://drive.google.com/open?id=0B0eJDPNaAh6BU1JNR2hNTFdSQUU

Here is the results of the commands:

```
$python -V
Python 2.7.6

 $ type module
module is a function
module () 
{ 
    eval $($LMOD_CMD bash "$@");
    [ $? = 0 ] && eval $(${LMOD_SETTARG_CMD:-:} -s sh)
}

 $ type -f module
bash: type: module: not found

 $ module av EasyBuild

----------------------------- /home/adrian/Apps/easybuild/modules/all -----------------------------
   EasyBuild/2.7.0 (L)

  Where:
   L:  Module is loaded

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".

 $ which -a eb
/home/adrian/Apps/easybuild/software/EasyBuild/2.7.0/bin/eb

 $ eb --version
This is EasyBuild 2.7.0 (framework: 2.7.0, easyblocks: 2.7.0) on host adrian-OptiPlex-780.
```

Thanks for your help.

**Comment by RalphTheBest** [comment]
In spite of the test still failing, I tried to make an installation to see if I could find anything else helpful. I tried to install hdf5 with the goolf toolchain using the command `eb --software-name=HDF5 --toolchain=goolf,1.4.10 -r`. As expected, it was unsuccessful and I'm unable to see why.
I'm leaving the stdout and the log, just in case it may be useful:
**Stdout**
https://drive.google.com/open?id=0B0eJDPNaAh6Bc2ZwN0x0NUprdVU
**Log**
https://drive.google.com/open?id=0B0eJDPNaAh6Bd3VHalJMNTdVNzQ

**Comment by boegel** [comment]
@RalphTheBest As far as I can tell, the output/logs for the failing tests is incomplete (output looks like an empty file, log seems to be cut off early)...

W.r.t. HDF5: the installation failed during the installation of GCC with:

```
/usr/local/bin/ld: cannot find crti.o: No such file or directory
/usr/local/bin/ld: cannot find -lc
/usr/local/bin/ld: cannot find crtn.o: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [libgcc_s.so] Error 1
```

This looks very much like a known issue on Ubuntu, cfr. https://github.com/hpcugent/easybuild/wiki/OS-specific-notes#ubuntu-1110

Can you maybe upload stdout/logs for the still failing tests via https://gist.github.com? That would make it easier to access them...

**Comment by RalphTheBest** [comment]
Hello again and sorry for the long time without answer, I haven't been at work these last days.

I was worried that maybe after so many attempts I had some installation broken or incomplete in the system that would prevent me from successfully install it, so I made a new clean installation of my ubuntu system.

The thing is, I tried installing Lmod 6.1 and followedthe OS specific steps that you suggested. I installed again easybuild from scratch with the bootstrap script and I run the command: `python -m test.framework.suite`.

Results here (I tried to use gist as you said, but for the log was impossible. It's too massive and I was simply unable to put it into gist without freezing my navigator):
**Stdout**
https://gist.github.com/RalphTheBest/c7add79c5e34993d1a8f65a6c44c1481
**Log**
https://drive.google.com/open?id=0B0eJDPNaAh6BTldGcWEyRG1JVjA

Again, just in case, I tried to install hdf5 anyway with `eb --software-name=HDF5 --toolchain=goolf,1.4.10 -r` to see the results. Failed as expected. 
Results here:
**Stdout**
https://gist.github.com/RalphTheBest/876fbb10b9ba3d4cb4876eb25cd3c3b1
**Log**
https://gist.github.com/RalphTheBest/f905a52a71de595f78f1f1c239a6b5a4

Sorry for all the inconveniences, but I'm unable to understand what is failing in the easybuild installation when I'm just using the bootstrap script. I don't know what can be failing in my system or my environment.

**Comment by boegel** [comment]
@RalphTheBest I don't see any errors in the output of the tests? Only some skipped tests, but that's normal, it's because you're missing some optional dependencies that are only required for some specific features of EasyBuild. How many tests failed, there should be a line printing that at the end, which you didn't include in the gist with the test output?

W.r.t. the GCC installation failing when you're trying to install HDF5, have you looked at the guidelines at https://github.com/hpcugent/easybuild/wiki/OS-specific-notes#ubuntu-1110?

For the problem mentioned in your log now, i.e. this:

```
In file included from /usr/include/bits/errno.h:24:0,
                 from /usr/include/errno.h:35,
                 from ../../../libgcc/../gcc/tsystem.h:94,
                 from ../../../libgcc/libgcc2.c:29:
/usr/include/linux/errno.h:1:23: fatal error: asm/errno.h: No such file or directory
compilation terminated.
```

please take a look at what's being suggested at http://askubuntu.com/a/530250 (i.e., install `linux-libc-dev`).

**Comment by RalphTheBest** [comment]
After some hard investigation, I found out that the fail in the tests was not easybuild's fault. It was due to some customizations made in my ubuntu version. Actually the problem was solved after the OS specific notes, I just didn't realized because of the modified system I'm using.

Thanks for all your help. I'm closing the issue as I think the problem was solved and there is nothing else to discuss.

Thanks again and sorry for trouble you.
